Polyhedron/demo/Polyhedron/Viewer.cpp

```diff
@@ -1690,6 +1690,9 @@ void Viewer::setTotalPass(int p)
 
 void Viewer::messageLogged(QOpenGLDebugMessage msg)
 {
+  //filter out useless warning
+  if(msg.message().contains("is being recompiled"))
+    return;
```

Comment:
Before we silence that warning, I would like you to explain why the NVidia OpenGL implementation emits a warning.

Reply:
I've read some things about NVIDIA not liking when a vertex shader is used multiple times, other things about NVidia not liking when a fragment shader is not bound, and mostly ppl saying it is a very useless warning.

Reply:
Can you please give links? I think the links should be in a comment in the code.

Reply:
https://community.khronos.org/t/vertex-shader-in-program-2-is-being-recompiled-based-on-gl-state/76019

https://stackoverflow.com/questions/12004396/opengl-debug-context-performance-warning

Reply:
That seems indeed a warning that we can ignore. Please insert those two URLs in the comments of the code.